app/actors/hyrax/actors/file_set_actor.rb

```diff
@@ -110,12 +112,28 @@ def build_file_actor(relation)
           file_actor_class.new(file_set, relation, user)
         end
 
+        def io_decorator(file)
+          Hydra::Derivatives::IoDecorator.new(file, mime_for(file), label_for(file))
+        end
+
+        def mime_for(file)
+          if file.respond_to?(:content_type)
+            file.content_type
+          elsif file.respond_to?(:mime_type)
+            file.mime_type
+          elsif file.respond_to?(:path)
+            MIME::Types.type_for(File.extname(file.path)).first.content_type
+          end # else nil
+        end
+
+        # For the label, use the original_filename or original_name if it's there.
+        # If the file was imported via URL, parse the original filename.
+        # If all else fails, use the basename of the file where it sits.
         def label_for(file)
-          # For the label, use the original filename if it's there.
-          # If the file was imported via URL, parse the original filename
-          # If all else fails, use the basename of the file where it sits
-          if file.respond_to?(:original_filename)
+          if file.respond_to?(:original_filename) # e.g. ActionDispatch::Http::UploadedFile, CarrierWave::SanitizedFile
             file.original_filename
+          elsif file.respond_to?(:original_name) # e.g. Hydra::Derivatives::IoDecorator
```

Comment:
the `file` argument to `update_content()` and `create_content()` do not permit `IoDecorator` in the YARD doc.  Do we need this clause?